### PR TITLE
[Merged by Bors] - Refactorings for Rust 1.59

### PR DESCRIPTION
**Pull request by RageKnify**
This Pull Request fixes new lint errors and warnings introduced in rust 1.59

Each commit describes its changse.

### CODE REVIEW

**Comment by raskad** [comment]
@RageKnify I'm guessing the codecov job fails because the coverage reduction is so big? But the coverage difference doesn't make much sense to me. There should not really be a difference right?

**Review by Razican — CHANGES_REQUESTED** [review]
I don't like the idea of renaming the typed arrays. We use the spec names in order to easier see the direct relationship. The rest looks good.

File: boa_engine/src/builtins/mod.rs [hunk 1]
```diff
@@ -140,17 +140,17 @@ pub fn init(context: &mut Context) {
...
-        Int16Array,
-        Uint16Array,
-        Int32Array,
-        Uint32Array,
-        BigInt64Array,
-        BigUint64Array,
-        Float32Array,
-        Float64Array,
+        Int8,
+        Uint8,
```
[the remark this answers is not in the document]
Comment (jedel1043) [reply boa_engine/src/builtins/mod.rs:144]: I agree, having only `Int8` as a built-in name is pretty confusing.

File: boa_engine/src/builtins/array_buffer/mod.rs [hunk 2]
```diff
@@ -402,12 +402,12 @@ impl ArrayBuffer {
         // 2. Return false.
         matches!(
             t,
-            TypedArrayName::Int8Array
-                | TypedArrayName::Uint8Array
-                | TypedArrayName::Int16Array
-                | TypedArrayName::Uint16Array
-                | TypedArrayName::Int32Array
-                | TypedArrayName::Uint32Array
+            TypedArrayName::Int8
```
[the remark this answers is not in the document]
Comment (RageKnify) [reply boa_engine/src/builtins/array_buffer/mod.rs:405]: The lint complains that every variant has the same suffix, not that the name is also contained in the enum name.

[the remark this answers is not in the document]
Comment (RageKnify) [reply boa_engine/src/builtins/array_buffer/mod.rs:405]: But I can accept keeping it as it was and just ignoring the lint for this case.

**Review by HalidOdat — APPROVED** [review]

**Review by Razican — APPROVED** [review]

**Pull request closed without merging**